### PR TITLE
CORTX-32320: Update docs for upgrade to clarify how images are used

### PR DESCRIPTION
**Pull request by walterlopatka**
## Description
* Update README.md to explain that specifying any image causes all three CORTX images to be pulled.
* Update -h help for upgrade-cortx-cloud.sh to explain the same
* Update README.md to describe how to update a single deployment or statefulset only.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds new functionality)
- [ ] Breaking change (bug fix or new feature that breaks existing functionality)
- [ ] Third-party dependency update
- [x] Documentation additions or improvements
- [ ] Code quality improvements to existing code or test additions/updates

## Applicable issues

- This change fixes an issue: CORTX-32320

## CORTX image version requirements
N/A

## How was this tested?
* Documentation change only.
* Verified that commands specified in doc changes function as described (kubectl set image)

## Checklist

- [ ] The change is tested and works locally.
- [ ] New or changed settings in the solution YAML are documented clearly in the README.md file.
- [x] All commits are signed off and are in agreement with the [CORTX Community DCO and CLA policy](https://github.com/Seagate/cortx/blob/main/doc/dco_cla.md).

If this change requires newer CORTX or third party image versions:

- [ ] The `image` fields in [solution.example.yaml](../k8_cortx_cloud/solution.example.yaml) have been updated to use the required versions.
- [ ] The `appVersion` field of the [Helm chart](../charts/cortx/Chart.yaml) has been updated to use the new CORTX version.

If this change addresses a CORTX Jira issue:

- [x] The title of the PR starts with the issue ID (e.g. `CORTX-XXXXX:`)

-----
[View rendered README.md](https://github.com/walterlopatka/cortx-k8s/blob/CORTX-32320_update_upgrade_docs/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,15 +179,38 @@ If you have direct access to the underlying Kubernetes Nodes in your cluster, CO
...
-2. Run the upgrade script to patch the CORTX on Kubernetes deployments using an updated image _(:information_source: You will want to update the `TARGET_IMAGE` variable below to your desired image tag)_. The script will stop all CORTX Pods, update the Deployments and StatefulSets, and then re-start the Pods.
+To upgrade a previously deployed CORTX cluster, run the `upgrade-cortx-cloud.sh` script to patch the CORTX on Kubernetes deployments using an updated image _(:information_source: You will want to update the `TARGET_IMAGE` variable below to your desired image tag)_. The script will stop all CORTX Pods, update the Deployments and StatefulSets, and then re-start the Pods.
 
    ```bash
-   TARGET_IMAGE="ghcr.io/seagate/cortx-data:2.0.0-790"
+   TARGET_IMAGE="ghcr.io/seagate/cortx-data:2.0.0-834"
    ./upgrade-cortx-cloud.sh -s solution.yaml -i $TARGET_IMAGE
    ```
 
+Note: There are three separate CORTX images (cortx-data, cortx-rgw, and cortx-control).  By specifying any one of these images, all images will be updated to that same version.  For example, if the image ghcr.io/seagate/cortx-data:2.0.0-790 is specified, then:
```
Comment (osowski) [inline README.md:191]: I would update your image version here to match what you updated in line 187 above.

File: README.md [hunk 2]
```diff
@@ -179,15 +179,38 @@ If you have direct access to the underlying Kubernetes Nodes in your cluster, CO
...
+
+- ghcr.io/seagate/cortx-data:2.0.0-834 will be applied to the cortx-data containers
+- ghcr.io/seagate/cortx-rgw:2.0.0-834 will be applied to the cortx-server containers
+- ghcr.io/seagate/cortx-control:2.0.0-834 will be applied to the cortx-control and cortx-ha containers
+
+
+To update the image for a specific CORTX deployment or statefulset use `kubectl set image`:
```
Comment (osowski) [inline README.md:198]: I would capitalize Deployment and StatefulSet here.

File: k8_cortx_cloud/upgrade-cortx-cloud.sh [hunk 3]
```diff
@@ -53,6 +53,11 @@ Usage:
 Options:
     -h              Prints help information.
     -i <IMAGE>      REQUIRED. The name of the container image to upgrade to.
+                    This image may be any of the three CORTX Docker images
```
Comment (osowski) [inline k8_cortx_cloud/upgrade-cortx-cloud.sh:56]: Instead of "Docker images", use "container images" .

**Review by osowski — APPROVED** [review]
A couple minor nits, based on documentation continuity. But otherwise I'm fine either way.

File: README.md (hunk 2 above)
Comment (walterlopatka) [reply README.md:198]: I agree. Will do.

File: k8_cortx_cloud/upgrade-cortx-cloud.sh (hunk 3 above)
Comment (walterlopatka) [reply k8_cortx_cloud/upgrade-cortx-cloud.sh:56]: @osowski , will do.
@keithpine, good idea!  I'll add a separate story for that.

File: README.md [hunk 4]
```diff
@@ -179,15 +179,38 @@ If you have direct access to the underlying Kubernetes Nodes in your cluster, CO
...
 
    ```bash
-   TARGET_IMAGE="ghcr.io/seagate/cortx-data:2.0.0-790"
+   TARGET_IMAGE="ghcr.io/seagate/cortx-data:2.0.0-834"
    ./upgrade-cortx-cloud.sh -s solution.yaml -i $TARGET_IMAGE
    ```
 
+Note: There are three separate CORTX images (cortx-data, cortx-rgw, and cortx-control).  By specifying any one of these images, all images will be updated to that same version.  For example, if the image ghcr.io/seagate/cortx-data:2.0.0-790 is specified, then:
+
+- ghcr.io/seagate/cortx-data:2.0.0-834 will be applied to the cortx-data containers
```
[the remark this answers is not in the document]
Comment (walterlopatka) [reply README.md]: I agree.  Will do.

**Pull request merged**